drivers/spi/Kconfig.rpi_pico

```diff
@@ -9,3 +9,26 @@ config SPI_RPI_PICO_PIO
 	select PICOSDK_USE_CLAIM
 	help
 	  Enable driving SPI via PIO on the PICO
+
+if SPI_RPI_PICO_PIO
+
+config SPI_RPI_PICO_PIO_8_BIT
+	bool "PIO 8-bit support"
+	default y
+	help
+	  Enable any addtional code needed to
+	  support 8-bit transfers.
+
+config SPI_RPI_PICO_PIO_16_BIT
+	bool "PIO 16-bit support"
+	help
+	  Enable any addtional code needed to
+	  support 16-bit transfers.
+
+config SPI_RPI_PICO_PIO_32_BIT
+	bool "PIO 32-bit support"
+	help
+	  Enable any addtional code needed to
+	  support 32-bit transfers.
```

Comment:
I think the same consideration as for HALF_DUPLEX is required here as well.

Reply:
Alas, the bit size being transferred can be defined at either build time or at run time, and can be changed on the fly.  Since this is a software configuration topic rather than hardware, it seems to be this belongs in Kconfig.  Which brings me back to my earlier question:  Does it make sense to include this fine a level of control?  Are the savings worth the overhead in the source?

Reply:
nah indeed, you can get rid of such config options. Just build the driver with the 3 size support.

Reply:
@soburi @tbursztyka - Good enough.  Have a gander at the latest force push and see what you think (assuming all the checks pass ...).